x/stakeibc/keeper/hooks.go

```diff
@@ -2,90 +2,82 @@ package keeper
 
 import (
 	"fmt"
+	"strconv"
 
+	utils "github.com/Stride-Labs/stride/utils"
 	epochstypes "github.com/Stride-Labs/stride/x/epochs/types"
+	icqkeeper "github.com/Stride-Labs/stride/x/interchainquery/keeper"
+	icqtypes "github.com/Stride-Labs/stride/x/interchainquery/types"
 	"github.com/Stride-Labs/stride/x/stakeibc/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	bankTypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	icatypes "github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts/types"
 	ibctypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
 	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
 )
 
+// TODO: ensure all timeouts are less than the epoch length
+// TODO: add events from event manager, e.g.
+// ctx.EventManager().EmitEvents(sdk.Events{
+// 	sdk.NewEvent(
+// 		sdk.EventTypeMessage,
+// 		sdk.NewAttribute("hostZone", zoneInfo.ChainId),
+// 		sdk.NewAttribute("newAmountStaked", balance.String()),
+// 	),
+// })
+
 func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
 	// every epoch
 	k.Logger(ctx).Info(fmt.Sprintf("Handling epoch start %s %d", epochIdentifier, epochNumber))
-	if epochIdentifier == "stride_epoch" {
+	if epochIdentifier == epochstypes.STRIDE_EPOCH {
 		k.Logger(ctx).Info(fmt.Sprintf("Stride Epoch %d", epochNumber))
+
+		// NOTE: We could nest this under `if epochNumber%depositInterval == 0 {`
+		// -- should we?
+		// e.g. CreateDepositRecordsForDepositInterval
+		// Imagine it will be slightly cleaner to track state by epoch, rather than
+		// by DepositInterval
+		if epochNumber < 0 {
+			panic("epochs must be positive")
+		}
+		epochTracker := types.EpochTracker{
+			EpochIdentifier: epochIdentifier,
+			EpochNumber: uint64(epochNumber),
+		}
+		// deposit records *must* exist for this epoch
+		k.SetEpochTracker(ctx, epochTracker)
+		k.Logger(ctx).Info("Triggering deposits")
+		// Create a new deposit record for each host zone for the upcoming epoch
+		k.CreateDepositRecordsForEpoch(ctx, epochNumber)
```

Comment:
I think I'm just missing something, why do we need to create empty deposit records for each host zone in each epoch? is there a lookup that we want to make sure won't fail?